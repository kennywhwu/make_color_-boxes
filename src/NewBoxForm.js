import React, { Component } from 'react';
import './NewBoxForm.css';

class NewBoxForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      color: '',
      width: '50px',
      height: '50px'
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(evt) {
    this.setState({ [evt.target.name]: evt.target.value });
  }

  handleSubmit(evt) {
    evt.preventDefault();
    this.props.handleNewBox(this.state);
    this.setState({ color: '', width: '50px', height: '50px' });
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <label htmlFor="color">Background Color:</label>
        <input
          id="color"
          name="color"
          value={this.state.color}
          onChange={this.handleChange}
        />

        <label htmlFor="width">Width:</label>
        <input
          id="width"
          name="width"
          value={this.state.width}
          onChange={this.handleChange}
        />
        <label htmlFor="height">Height:</label>
        <input
          id="height"
          name="height"
          value={this.state.height}
          onChange={this.handleChange}
        />
        <button>Create a new box!</button>
      </form>
    );
  }
}

export default NewBoxForm;
