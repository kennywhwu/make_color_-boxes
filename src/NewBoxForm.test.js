import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NewBoxForm from './NewBoxForm';

// Smoke Test
it('renders without crashing', function() {
  shallow(<NewBoxForm />);
});

// Snapshot Test
it('matches snapshot', function() {
  let wrapper = shallow(<NewBoxForm />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

// // Simulate submit form
// it('can be submitted with expected output', function() {
//   let wrapper = shallow(<NewBoxForm />);

//   wrapper.setState({
//     color: 'red',
//     width: '1050px',
//     height: '50px'
//   });
//   wrapper.simulate('submit');
//   let i_elem = wrapper.find('i').first();
//   // note {42}, not just 42 --- has to match exactly
//   expect(i_elem.equals(<i>{43}</i>)).toBe(true);
// }); // end
