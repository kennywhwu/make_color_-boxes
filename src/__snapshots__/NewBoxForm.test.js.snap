// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="color"
  >
    Background Color:
  </label>
  <input
    id="color"
    name="color"
    onChange={[Function]}
    value=""
  />
  <label
    htmlFor="width"
  >
    Width:
  </label>
  <input
    id="width"
    name="width"
    onChange={[Function]}
    value="50px"
  />
  <label
    htmlFor="height"
  >
    Height:
  </label>
  <input
    id="height"
    name="height"
    onChange={[Function]}
    value="50px"
  />
  <button>
    Create a new box!
  </button>
</form>
`;
